src/btree/bt_split.c

```diff
@@ -1720,49 +1721,89 @@ __split_insert(WT_SESSION_IMPL *session, WT_REF *ref)
 	 */
 	ref->addr = NULL;
 
-	/*
-	 * Copy the first key from the original page into first ref in the new
-	 * parent.  Pages created in memory always have a "smallest" insert
-	 * list, so look there first.  If we don't find one, get the first key
-	 * from the disk image.
-	 *
-	 * We can't just use the key from the original ref: it may have been
-	 * suffix-compressed, and after the split the truncated key may not be
-	 * valid.
-	 */
-	WT_ERR(__wt_scr_alloc(session, 0, &key));
-	if ((ins = WT_SKIP_FIRST(WT_ROW_INSERT_SMALLEST(page))) != NULL) {
-		key->data = WT_INSERT_KEY(ins);
-		key->size = WT_INSERT_KEY_SIZE(ins);
+	if (type == WT_PAGE_ROW_LEAF) {
+		/*
+		 * Copy the first key from the original page into first ref in
+		 * the new parent. Pages created in memory always have a
+		 * "smallest" insert list, so look there first.  If we don't
+		 * find one, get the first key from the disk image.
+		 *
+		 * We can't just use the key from the original ref: it may have
+		 * been suffix-compressed, and after the split the truncated key
+		 * may not be valid.
+		 */
+		WT_ERR(__wt_scr_alloc(session, 0, &key));
+		if ((ins =
+		    WT_SKIP_FIRST(WT_ROW_INSERT_SMALLEST(page))) != NULL) {
+			key->data = WT_INSERT_KEY(ins);
+			key->size = WT_INSERT_KEY_SIZE(ins);
+		} else
+			WT_ERR(__wt_row_leaf_key(
+			    session, page, &page->pg_row_d[0], key, true));
+		WT_ERR(__wt_row_ikey(session, 0, key->data, key->size, child));
+		parent_incr += sizeof(WT_IKEY) + key->size;
+		__wt_scr_free(session, &key);
 	} else
-		WT_ERR(__wt_row_leaf_key(
-		    session, page, &page->pg_row_d[0], key, true));
-	WT_ERR(__wt_row_ikey(session, 0, key->data, key->size, child));
-	parent_incr += sizeof(WT_REF) + sizeof(WT_IKEY) + key->size;
-	__wt_scr_free(session, &key);
+		child->key.recno = ref->key.recno;
+
+	/*
+	 * Don't copy any deleted page state: we may be splitting a page that
+	 * was instantiated after a truncate and that history should not be
+	 * carried onto these new child pages.
+	 */
+	child->page_del = NULL;
```

Comment:
@michaelcahill 
> How could this be set? We start with zeroed memory and only set the fields we want.
I understand wanting the comment, maybe it can go near line 1716?

Agreed, fixed.